src/chrome/extension/scripts/background.ts

```diff
@@ -41,15 +44,33 @@ chrome.runtime.onMessageExternal.addListener(
         return true;
     });
 
-// Launch the Chrome webstore page for the uProxy app.
+// Launch the Chrome webstore page for the uProxy app,
+// or activate the user's tab open to uproxy.org/chrome-install
 function openDownloadAppPage() : void {
-  chrome.tabs.create(
-      {url: 'https://chrome.google.com/webstore/detail/uproxyapp/fmdppkkepalnkeommjadgbhiohihdhii'},
-      (tab) => {
-        // Focus on the new Chrome Webstore tab.
-        chrome.windows.update(tab.windowId, {focused: true});
+  chrome.windows.get(mainWindowId, {populate: true},
+      (windowThatLaunchedUproxy) => {
+        if (windowThatLaunchedUproxy) {
+          for (var i = 0; i < windowThatLaunchedUproxy.tabs.length; i++) {
+            // If the user is installing via the inline install flow,
+            // instead of sending them to the webstore to install the app,
+            // bring them back to uproxy.org/chrome-install
+            if ((windowThatLaunchedUproxy.tabs[i].url.indexOf("uproxysite.appspot.com/chrome-install") > -1) ||
+                (windowThatLaunchedUproxy.tabs[i].url.indexOf("uproxy.org/chrome-install") > -1)) {
+              chrome.tabs.update(windowThatLaunchedUproxy.tabs[i].id, {active:true});
+              chrome.windows.update(mainWindowId, {focused: true});
+              return;
+            }
+          }
+        }
+        // Only reached if the user didn't have uproxy.org/chrome-install open.
+        chromeCoreConnector.waitingForAppInstall = true;
+        chrome.tabs.create(
+            {url: 'https://chrome.google.com/webstore/detail/uproxyapp/fmdppkkepalnkeommjadgbhiohihdhii'},
```

Comment:
We are assuming if the user does not have uproxy.org/chrome-install open, then they are installing via the webstore install flow. In that case, we send them to the webstore for consistency sake. Hmm do you think this is too strong an assumption?